geps/gep-713.md

```diff
@@ -38,24 +58,210 @@ portable.
 * Define all potential policies that may be attached to resources
 * Design the full structure and configuration of policies
 
+## Background and concepts
+
+When designing Gateway API, one of the things we’ve found is that we often need to be
+able change the behavior of objects without being able to make changes to the spec
+of those objects. Sometimes, this is because we can’t change the spec of the object
+to hold the information we need ( ReferenceGrant, from GEP-709, affecting Secrets
+and Services is an example, as is Direct Policy Attachment), and sometimes it’s
+because we want the behavior change to flow across multiple objects
+(this is what Hierarchical Policy Attachment is for).
+
+To put this another way, sometimes we need ways to be able to affect how an object
+is interpreted in the API, without representing the description of those effects
+inside the spec of the object.
+
+This document describes the ways we design objects to meet these two use cases,
+and why you might choose one or the other.
+
+We use the term “metaresource” to describe the class of objects that _only_ augment
+the behavior of another Kubernetes object, regardless of what they are targeting.
+
+“Meta” here is used in its Greek sense of “more comprehensive”
+or “transcending”, and “resource” rather than “object” because “metaresource”
+is more pronounceable than “metaobject”. Additionally, a single word is better
+than a phrase like “wrapper object” or “wrapper resource” overall, although both
+of those terms are effectively synonymous with “metaresource”.
+
+A "Policy Attachment" is a metaresource that affects the fields in existing objects
+(like Gateway or Routes), or influences the configuration that's generated in an
+underlying data plane.
+
+"Direct Policy Attachment" is when a Policy object references a single object _only_,
+and only modifies the fields of or the configuration associated with that object.
+
+"Hierarchical Policy Attachment" is when a Policy object references a single object
+_and any child objects of that object_ (according to some defined hierarchy), and
```

Comment:
I'd agree with that, but we actually use at least two different hierarchies in the examples... (I'm also using Namespace -> Gateway, since we suggest _not_ using GatewayClass as a way to apply Policy except in certain circumstances).

I'd prefer to leave this as "some defined" because then it covers something like having a Policy that controls a custom HTTP Filter, which is not _exactly_ the same as the GatewayClass -> Gateway -> HTTPRoute -> Backend hierarchy we will _mainly_ be using.